bika/lims/content/analysisrequest.py

```diff
@@ -2357,6 +2363,30 @@ def getVerifier(self):
                 verifier = actor
         return verifier
 
+    @security.public
+    def getVerifiersIDs(self):
+        """Returns the ids from users that have verified at least one analysis
+        from this Analysis Request
+        """
+        verifiers_ids = list()
+        for brain in self.getAnalyses():
+            verifiers = brain.getVerificators or ""
+            verifiers_ids += verifiers.split(",")
+        return list(set(verifiers_ids))
+
+    @security.public
+    def getVerifiers(self):
+        """Returns the list of lab contacts that have verified at least one
+        analysis from this Analysis Request
+        """
+        contacts = list()
+        for verifier in self.getVerifiersIDs():
+            user = api.get_user(verifier)
+            contact = api.get_user_contact(user, ["LabContact"])
+            if contact:
+                contacts.append(contact)
+        return contacts
```

Comment:
You simply skip here user ids which are not bound to a lab contact. Shouldn't we keep at least the ID to display?

Reply:
I am skipping them because I think lab users not binded to a lab-contact should never exist, except from "admin".

Reply:
Anyhow, you'll get the ID if you call "getVerifiersIDs". But this function returns "LabContact" types precisely...

Reply:
> I am skipping them because I think lab users not binded to a lab-contact should never exist, except from "admin".

Yes indeed, but unfortunately **we** didn't manage to bind lab contacts to senaite users in our test instance yet...

Reply:
@grulisco: In your case this is managed in your custom report, so this code has no side-effect.